transport/src/main/java/io/netty/channel/ChannelHandlerMask.java

```diff
@@ -163,8 +168,34 @@ private static boolean isSkippable(
             final Class<?> handlerType, final String methodName, final Class<?>... paramTypes) throws Exception {
 
         return AccessController.doPrivileged((PrivilegedExceptionAction<Boolean>) () ->
-                handlerType.getMethod(methodName, paramTypes).isAnnotationPresent(ChannelHandler.Skip.class));
+                handlerType.getMethod(methodName, paramTypes).isAnnotationPresent(Skip.class));
     }
 
     private ChannelHandlerMask() { }
+
+    /**
+     * Indicates that the annotated event handler method in {@link ChannelHandler} will not be invoked by
+     * {@link ChannelPipeline}. This annotation is only useful when your handler method implementation
+     * only passes the event through to the next handler, like the following:
+     *
+     * <pre>
+     * {@code @Skip}
+     * {@code @Override}
+     * public void channelActive({@link ChannelHandlerContext} ctx) {
+     *     ctx.fireChannelActive(); // do nothing but passing through to the next handler
+     * }
+     * </pre>
+     *
+     * <p>
+     * Note that this annotation is not {@linkplain Inherited inherited}.  If you override a method annotated with
+     * {@link Skip}, it will not be skipped anymore.  Similarly, you can override a method not annotated with
```

Comment:
@normanmaurer I guess this removes the ability to re-skip methods for whatever reason in subclasses of handlers that implement them as described here? It's a fairly specific and presumably rare situation so maybe keeping things simple as you have done here out-weighs any slight benefit.

But then maybe it's best to update the javadoc here (it's also generally worded in a way that sounds like it's for external consumption)?

Reply:
@njhill yeah I think it is fine to not let people do this as it can also lead to some very hard to debug problems if you forget to remove the annotation.

Reply:
@njhill reworded and pushed. Let me know what you think